misc/coding.md

```diff
@@ -555,3 +438,29 @@ The post-increment and post-decrement operators create an unnecessary copy, that
 ### Alternative Operator Representations
 
 Use the primary token for operators (E.g. `&&`, `||`, and `!`), alternative representations are not allowed (`and`, `or`, and `not`).
+
+## Index naming convention
+
+To have an uniform usage of names for indices and sizes in the library, please follow the following conventions:
+- Use `m`, `n` and `k` for sizes and `i`, `j` and `l`,
+- `i` and `m` in general refers to rows (exceptions square sizes and diagonal),
+- `j` and `n` in general refers to columns (exceptions square sizes and diagonal).
+
+Note: if the use of the row or column index/size is determined by a template or a function parameter `i`, `m` should be used for the "parallel" coordinate and `j`, `n` for the orthogonal one.
+
+2D Indices:
+- Use the two indices to name the 2D object. E.g. `ij`. Note: In this way extracting a single index with `SizeType i = ij.row();` is intuitive.
+- The use of `id`, `idx` and similar names is discouraged.
+2D Size:
+- Name the object to identify what it refers to. E.g. `size_a` represent the number of global tiles of the matrix A.
```

Comment:
In this case I decided to not use combined names as they are hard to read. E.g. I find `mn` hard to interpret...